pkg/engine/armvariables.go

```diff
@@ -74,6 +74,11 @@ func getK8sMasterVars(cs *api.ContainerService) (map[string]interface{}, error)
 	isCustomVnet := cs.Properties.AreAgentProfilesCustomVNET()
 	hasAgentPool := len(profiles) > 0
 	hasCosmosEtcd := masterProfile != nil && to.Bool(masterProfile.CosmosEtcd)
+	
+	kubernetesVersion := orchProfile.OrchestratorVersion
+	if cs.Properties.IsAzureStackCloud() {
```

Comment:
can we add a unit test for this?

Reply:
I fixed an existing test that I think covers the new execution flow.